src/app/tests/DevToys.UnitTests/Core/Tools/SmartDetectionServiceTests.cs

```diff
@@ -9,8 +9,8 @@ public class SmartDetectionServiceTests : MefBasedTest
     [Theory]
     [InlineData("hello world", true, "MockTool2")]
     [InlineData("hello world", false, "MockTool2")]
-    [InlineData("{ \"json\": 123 }", true, "MockTool")]
-    [InlineData("{ \"json\": 123 }", false, "MockTool", "MockTool2")]
+    [InlineData("{ \"json\": 123 }", true, "MockTool2")]
```

Comment:
This doesn't fix the test. We do expect `MockTool` on line 12, and `MockTool and MockTool2` on line 13.

The reason why this test is broken is because the class `MockJsonDataTypeDetector` is using `System.Text.Json` with some bad configuration.

To fix the test, revert this change, then modify `MockJsonDataTypeDetector.TryDetectDataAsync` to have a similar behavior to the `JsonDataTypeDetector` you made:

```
[Export(typeof(IDataTypeDetector))]
[DataTypeName("JSON", baseName: "text")]
internal class MockJsonDataTypeDetector : IDataTypeDetector
{
    public ValueTask<DataDetectionResult> TryDetectDataAsync(object data, DataDetectionResult resultFromBaseDetector, CancellationToken cancellationToken)
    {
        if (resultFromBaseDetector is not null
            && resultFromBaseDetector.Data is string dataString
            && !long.TryParse(dataString, out _))
        {
            if (JsonHelper.IsValid(dataString))
            {
                return ValueTask.FromResult(new DataDetectionResult(Success: true, Data: dataString));
            }
        }

        return ValueTask.FromResult(DataDetectionResult.Unsuccessful);
    }
}
```